mono/mini/method-to-ir.c

```diff
@@ -6586,6 +6768,7 @@ mono_method_to_ir (MonoCompile *cfg, MonoMethod *method, MonoBasicBlock *start_b
 		   guint inline_offset, gboolean is_virtual_call)
 {
 	ERROR_DECL (error);
+	int safe_auto_tailcall_cache = 0;
```

Comment:
Can this be safely/efficiently/more-efficiently be moved into MonoMethod?
1. 2 bits is enough, but might not be atomic, so int?
To some extend non-atomicity is ok, in a race, multiple threads could recompute it, but I worry if it is bitfield, we could have read/update/write and it could tear other bits. Is an int too much to add to every MonoMethod?

2. Can a MonoMethod be compiled multiple times concurrently and then race at commit time?
If not, then a 2 bit bitfield suffices.
However opacity would suggest a typedef and a typedef could be only as small as a char.
Since it is every MonoMethod, I'm nervous.

3. The reason to put in MonoMethod is, well, it doesn't matter currently, but if we can tailcall within an inlined method, that calculation can be reused across multiple inlinings and non-inlined version. Currently I never tailcall while inlining, but that should go away.